server/src/main/java/io/crate/replication/logical/LogicalReplicationService.java

```diff
@@ -144,56 +151,66 @@ private void handleRepositoriesForChangedSubscriptions(@Nullable SubscriptionsMe
-                repositoriesService.unregisterInternalRepository(REMOTE_REPOSITORY_PREFIX + subscriptionName);
+            boolean subscriptionRemoved = newSubscriptions.get(subscriptionName) == null;
+            if (subscriptionRemoved) {
+                removeSubscription(subscriptionName);
             }
-
             // Close connection for removed subscription (must not run inside a transport thread)
             withRetry(() -> updateRemoteConnection(subscriptionName, Settings.EMPTY));
         }
     }
 
+    private void addSubscription(String subscriptionName, Subscription subscription) {
+        assert repositoriesService != null
+            : "RepositoriesService must be set immediately after LogicalReplicationService construction";
+        LOGGER.debug("Adding new logical replication repository for subscription '{}'", subscriptionName);
+        repositoriesService.registerInternalRepository(REMOTE_REPOSITORY_PREFIX + subscriptionName, TYPE);
+
+        if (clusterService.localNode().isMasterNode()) {
+            withRetry(
+                () -> {
+                    // The startReplication will initiate the remote connection upfront
+                    LOGGER.debug("Start logical replication for subscription '{}'", subscriptionName);
+                    startReplication(subscriptionName, subscription, new ActionListener<>() {
+                        @Override
+                        public void onResponse(AcknowledgedResponse acknowledgedResponse) {
+                            LOGGER.debug("Acknowledged logical replication for subscription '{}'", subscriptionName);
+                            LOGGER.debug("Start tracking metadata for subscription '{}'", subscriptionName);
+                            synchronizeTableDefinitions.addSubscriptions(subscriptionName);
+                        }
+
+                        @Override
+                        public void onFailure(Exception e) {
+                            LOGGER.debug("Failure for logical replication for subscription", e);
+                        }
+                    });
+                }
+            ).run();
+        } else {
+            // Initiate the remote connection for the subscription, needed on all nodes to restore the
+            // repository and for following remote shard changes.
+            // Must run inside a transport thread
+            withRetry(() -> updateRemoteConnection(subscriptionName)).run();
+        }
+    }
+
+    private void removeSubscription(String subscriptionName) {
+        assert repositoriesService != null
+            : "RepositoriesService must be set immediately after LogicalReplicationService construction";
+        LOGGER.debug("Removing logical replication repository for dropped subscription '{}'",
+                     subscriptionName);
+        repositoriesService.unregisterInternalRepository(REMOTE_REPOSITORY_PREFIX + subscriptionName);
+        // Stop tracking metadata for a removed subscription
```

Comment:
```suggestion
```

The comment just repeats the code below.

Question would be if the naming doesn't fit given that the comment used different wording.

E.g should it be `metadataTracker.stopTracking(subscriptionName)` or something like that?

Reply:
👍